### DISCUSSION TITLE
remove state: templating

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [Mariusthvdb/custom-ui#53] [Ideas]
Since we we can still template/customize state: in custom-ui, I would suggest we take that  out.

There a several reasons for that:

- it can have unexpected, and fundamental effects in HA state machine, possibly even fatal. (with it, we truly change the state of an entity, and not merely the representation of the state)
- it was a seldomly used feature, so taking it out relieves maintaining it
- it can easily be replaced by modern custom-cards in Lovelace (which the frontend representation of `state:` is doing after all, please use custom:template-entity-row by Thomas Loven, or state_display options I custom:button-card by Romrider
- The former stateDisplay was already taken out, this last edit would completely make custom-ui work solely in Lovelace, and safe at that.

Even on the original custom-ui repo Andrey already [said](https://github.com/andrey-git/home-assistant-custom-ui/blob/master/docs/templates.md#special-attributes) to be careful.

**Comment by bratanon** [comment]
I agree on this. And it currently broken as of the upgrade of HA 2022.4.

I have created a [PR](https://github.com/Mariusthvdb/custom-ui/pull/54) which does not handle state templates, and warns the users that are using it with a console warning. 

Users that still have this in there config will not have any braking issues except that the state wont change as intendent anymore.

**Comment by nolenjohan** [comment]
So this will no longer work? To get different colors depending on state (in this case temperatures):

  templates:
    icon_color: >
      if (state > 25 ) return 'rgb(178, 31, 31)';
      if (state >= 20 ) return 'rgb(220, 130, 40)';
      if (state < 0 ) return 'rgb(0, 0, 255)';
      if (state < 5 ) return 'rgb(0, 70, 209)';
      if (state < 10 ) return 'rgb(0, 139, 209)';
      return 'rgb(249, 228, 6)';

**Comment by Mariusthvdb** [comment]
no, that will ofc still work.

we stopped supporting customizing the state, 
```
templates:
  state:
```

**Comment by fn87** [comment]
So this should work?

`group.perimeter:
  icon: mdi:exit-run
  templates:
    icon: if(state === 'on') return 'mdi:exit-run'; return 'mdi:door-closed';
    icon_color: if (state === 'off') return 'green'; return 'red';
`
`group.movement:
  templates:
    icon: >
      if(state === 'on') return 'mdi:run-fast'; 
      return 'mdi:human-male';
    icon_color: if (state === 'off') return 'green'; return 'red';`

I need to refresh page for the templates to take effect.

**Reply by Mariusthvdb** [reply]
Please update to the latest release.

If you have an issue, open an issue .
